### PR TITLE
Update "Explicit MFA Deny"

### PR DESCRIPTION
**Pull request by tduarte14**
Required items, please complete
   
   Change(s):
   - Remove redundant line declaring Type (it's already there)
   - Add Host and OS fields to the end of the query
   - Add extra entities: App and Host
   - Fix IP entity (was using inexistent field)
   - Removed URL entity (was using ClientAppUsed, which was giving no value)

   Reason for Change(s):
   - Some Entities had broken field mapping, another had no value and some were needed.

   Version Updated:
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml [hunk 1]
```diff
@@ -49,14 +49,20 @@ entityMappings:
...
-        columnName: IPAddress
+        columnName: PublicIP
   - entityType: AzureResource
     fieldMappings:
-      - identifier: ResourceID
-        columnName: ResourceID
-  - entityType: URL
+      - identifier: ResourceId
+        columnName: ResourceId
+  - entityType: CloudApplication
```
Comment (ashwin-patil) [inline Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml:57]: can you also add  identifier: AppId and Fieldmapping for CloudApplication entity.

**Review by ashwin-patil — CHANGES_REQUESTED** [review]
suggested couple of changes on entity mappings

File: Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml [hunk 2]
```diff
@@ -24,21 +24,21 @@ relevantTechniques:
...
-  | project PublicIP, AadDeviceId; 
+    | extend SensorHealthState = column_ifexists("SensorHealthState", "")
+    | where OnboardingStatus == "Onboarded" and SensorHealthState == "Active" 
+    | project PublicIP, AadDeviceId; 
   let aadNonInt = aadFunc("AADNonInteractiveUserSignInLogs"); 
   union isfuzzy=true aadSignin, aadNonInt 
   | project-rename PublicIP = IPAddress   
-  | join kind=leftouter dvcInfo on PublicIP  
+  | join kind=leftouter dvcInfo on PublicIP
+  | extend Host=tostring(DeviceDetail_dynamic.displayName), OS=tostring(DeviceDetail_dynamic.operatingSystem)
```
[the remark this answers is not in the document]
Comment (tduarte14) [reply Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml:41]: The thing is that the same Public IP can be related to different devices, so I'm not sure you can even have a correct match from that block of code there. The host I got was from the signin logs, which is what we actually see on the logs. DeviceInfo shows many times no publicIP nor logged user, so it's hard to have a match.

**Review by tduarte14 — COMMENTED** [review]
added comments

File: Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tduarte14) [reply Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml:41]: If you please don't agree, please review again.

**Review by tduarte14 — COMMENTED** [review]
added comments as why I don't agree on the review. Please check

**Comment by tduarte14** [comment]
Hi any feedback here?

**Comment by v-prasadboke** [comment]
Hello @tduarte14 Ashwin is on leave till 22. He will get back to you as he comes back.

**Comment by ashwin-patil** [comment]
> Hi any news here?

Hey, i started the review but forgot to submit final comments. Apologies for delay, let me know your thoughts on follow-up comments. I will keep an eye on this PR this week to respond sooner.

**Comment by v-prasadboke** [comment]
Hey Thank you for the updates @ashwin-patil.

**Comment by v-prasadboke** [comment]
Hello @tduarte14, Ashwin has requested some changes please work on it

**Comment by v-prasadboke** [comment]
Hello @tduarte14,
Ashwin has suggested some changes please take a look into it.

**Comment by v-prasadboke** [comment]
Hello @tduarte14, 
Ashwin has requested some changes earlier, Please have a look on it.

**Comment by v-prasadboke** [comment]
Hello @tduarte14, 
You might be busy with something, commenting on this PR to please acknowledge Ashwin's comments.

**Comment by tduarte14** [comment]
Got the latest version 1.0.5, removed URL entity and added just CloudApplication entity. Ignored the Host entity for now

**Comment by tduarte14** [comment]
Also changed runtime to every 1h instead of 1d to improve MTTD. Added missing appId field on CloudApplication entity as you requested, thought it was there already

**Comment by v-prasadboke** [comment]
Thank you @tduarte14.

**Review by ashwin-patil — APPROVED** [review]

File: Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwin-patil) [reply Solutions/Microsoft Entra ID/Analytic Rules/ExplicitMFADeny.yaml]: Sounds good @tduarte14 . We will revisit this sometime again later. 
Thanks for the feedback.

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**